encoders/nlp/UniversalSentenceEncoder/__init__.py

```diff
@@ -26,21 +81,25 @@ def __init__(
         :param kwargs:
         """
         super().__init__(*args, **kwargs)
+
         self.model_url = model_url
 
+        if self.model_url == MODEL_ENCODER_CMLM:
+            self.sentence_encoder = CMLMEncoder()
+        else:
+            self.sentence_encoder = GeneralEncoder(self.model_url)
+
     def post_init(self):
         self.to_device()
-        import tensorflow_hub as hub
-        self.model = hub.load(self.model_url)
+        self.sentence_encoder.post_init()
```

Comment:
You should directly instantiatr the private classes here, otherwise they will be serialized. Also no need to have post_init pattern in the private classes.